src/native/minipal/random.c

```diff
@@ -103,6 +107,52 @@ int32_t minipal_get_cryptographically_secure_random_bytes(uint8_t* buffer, int32
     static volatile int rand_des = -1;
     static bool sMissingDevURandom;
 
+#if HAVE_GETRANDOM
+    static bool sMissingGetRandomSysCall = false;
+
+    if (!sMissingGetRandomSysCall)
```

Comment:
```suggestion
    static short sGetRandomSupport = -1;
    if (sGetRandomSupport == -1)
    {
        sGetRandomSupport = syscall(SYS_getrandom, nullptr, 0, GRND_NONBLOCK) == -1 && errno == ENOSYS ? 0 : 1;
    }

    if (sGetRandomSupport == 1)
```

Reply:
Why is this better than using the `getrandom` provided by the C runtime?

Reply:
That could work. C runtime can wrap it and may hide ENOSYS while syscall is explicit, we used it in numasupport and few other places.

Reply:
The direct syscall has a small advantage in that it does not introduce hard dependency on the new glibc version, and so the binaries work in practice on older baselines too.

For example, even though are current baseline is Ubuntu 16, the binaries work in practice on older baselines too since we do not actually have a hard dependency on anything from Ubuntu 16 (it has been verified for native AOT compiled binaries, I am not sure about full coreclr).